pkg/cloudprovider/aws/instancetype.go

```diff
@@ -60,66 +60,100 @@ func (i *InstanceType) Architecture() string {
 	return fmt.Sprint(aws.StringValueSlice(i.ProcessorInfo.SupportedArchitectures)) // Unrecognized, but used for error printing
 }
 
-func (i *InstanceType) CPU() *resource.Quantity {
-	return resources.Quantity(fmt.Sprint(*i.VCpuInfo.DefaultVCpus))
+func (i *InstanceType) Resources() v1.ResourceList {
+	return v1.ResourceList{
+		v1.ResourceCPU:             i.cpu(),
+		v1.ResourceMemory:          i.memory(),
+		v1.ResourcePods:            i.pods(),
+		v1alpha1.ResourceAWSPodENI: i.awsPodENI(),
+		v1alpha1.ResourceNVIDIAGPU: i.nvidiaGPUs(),
+		v1alpha1.ResourceAMDGPU:    i.amdGPUs(),
+		v1alpha1.ResourceAWSNeuron: i.awsNeurons(),
+	}
+}
+
+func (i *InstanceType) Price() float64 {
+	const (
+		GPUCostWeight      = 5
+		CPUCostWeight      = 1
+		MemoryMBCostWeight = 1
+		MbToGb             = 1 / 1024.0
```

Comment:
Thoughts on modeling this in MB?
```suggestion
		MemoryMBCostWeight = 1024
```